packages/x-components/src/x-modules/facets/components/preselected-filters.vue

```diff
@@ -1,69 +1,81 @@
 <script lang="ts">
-  import { Component, Inject, Prop, Watch } from 'vue-property-decorator';
-  import Vue from 'vue';
+  import { defineComponent, inject, PropType, onMounted, watch, computed, ComputedRef } from 'vue';
   import { createRawFilters } from '../../../utils/filters';
   import { isArrayEmpty } from '../../../utils/array';
   import { SnippetConfig } from '../../../x-installer/api/api.types';
+  import { use$x, useNoElementRender } from '../../../composables/index';
 
   /**
    * This component emits {@link FacetsXEvents.PreselectedFiltersProvided} when a preselected filter
    * is set in the snippet config or by using the prop of the component.
    *
    * @public
    */
-  @Component
-  export default class PreselectedFilters extends Vue {
-    /**
-     * Injects {@link SnippetConfig} provided by an ancestor as snippetConfig.
-     *
-     * @internal
-     */
-    @Inject('snippetConfig')
-    public snippetConfig?: SnippetConfig;
-
-    /**
-     * A list of filters to preselect.
-     *
-     * @remarks Emits the {@link FacetsXEvents.PreselectedFiltersProvided} when the
-     * component is created.
-     *
-     * @public
-     */
-    @Prop({ default: () => [] })
-    public filters!: string[];
-
-    /**
-     * Gets the provided preselected filters prioritizing the {@link SnippetConfig} over the
-     * filters prop.
-     *
-     * @returns An array of filter's ids.
-     */
-    protected get preselectedFilters(): string[] {
-      return this.snippetConfig?.filters ?? this.filters;
-    }
-
-    /**
-     * Emits the {@link FacetsXEvents.PreselectedFiltersProvided} to save
-     * the provided filters in the state.
-     */
-    @Watch('preselectedFilters')
-    protected emitPreselectedFilters(): void {
-      if (!isArrayEmpty(this.preselectedFilters)) {
-        this.$x.emit('PreselectedFiltersProvided', createRawFilters(this.preselectedFilters));
+  export default defineComponent({
+    name: 'PreselectedFilters',
+    props: {
+      /**
+       * A list of filters to preselect.
+       *
+       * @remarks Emits the {@link FacetsXEvents.PreselectedFiltersProvided} when the
+       * component is created.
+       *
+       * @public
+       */
+      filters: {
+        type: Array as PropType<string[]>,
+        default: () => []
       }
+    },
+    setup(props, { slots }) {
+      const $x = use$x();
```

Comment:
use the composable for the bus instead of the $x

Reply:
Hi! Using the `useXBus` composable leads to a Vue warn (injection "location" not found) in this use case, maybe I am using it the wrong way?

Reply:
It's just a warning and the location can be `undefined` anyway, but it makes more sense that the default value for the location is `'none'`, so change the `useXBus` composable so it has a default value for the location.

```
const location = inject<FeatureLocation>('location', 'none');
```